underpants.go

```diff
@@ -111,6 +127,7 @@ func (u *user) encode(key []byte) (string, error) {
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
+  w.Close()
```

Comment:
It captures the same use case as C++ RAII destructors. So as soon as you have secured a resource, you can use defer to run that code when the function exits.
`
w := base64.NewEncoder(base64.URLEncoding,
     io.MultiWriter(h, &b))
defer w.Close()
`

The benefits in this case aren't all that great, but defer does help with code evolution since future authors do not have to analyze all return paths to determine where to close a resource.